reconcile/templating/renderer.py

```diff
@@ -0,0 +1,172 @@
+import os
+from abc import ABC, abstractmethod
+from collections.abc import Callable
+from typing import Any, Optional
+
+from pydantic import BaseModel
+from ruamel import yaml
+
+from reconcile.gql_definitions.templating.template_collection import (
+    TemplateCollectionV1,
+    TemplateCollectionVariablesV1,
+    TemplateV1,
+    query,
+)
+from reconcile.templating.rendering import (
+    TemplateData,
+    create_renderer,
+)
+from reconcile.utils import gql
+from reconcile.utils.gql import init_from_config
+from reconcile.utils.runtime.integration import (
+    PydanticRunParams,
+    QontractReconcileIntegration,
+)
+
+QONTRACT_INTEGRATION = "template-renderer"
+
+APP_INTERFACE_PATH_SEPERATOR = "/"
+
+
+def get_template_collections(
+    query_func: Optional[Callable] = None,
+) -> list[TemplateCollectionV1]:
+    if not query_func:
+        query_func = gql.get_api().query
+    return query(query_func).template_collection_v1 or []
+
+
+class TemplateOutput(BaseModel):
+    path: str
+    content: str
+
+
+class FilePersistence(ABC):
+    @abstractmethod
+    def write(self, outputs: list[TemplateOutput]) -> None:
+        pass
+
+    @abstractmethod
+    def read(self, path: str) -> str:
+        pass
+
+
+class LocalFilePersistence(FilePersistence):
+    def __init__(self, app_interface_data_path: str) -> None:
+        self.app_interface_data_path = app_interface_data_path
+
+    def write(self, outputs: list[TemplateOutput]) -> None:
+        for output in outputs:
+            with open(
+                f"{join_path(self.app_interface_data_path, output.path)}",
+                "w",
+                encoding="utf-8",
+            ) as f:
+                f.write(output.content)
+
+    def read(self, path: str) -> str:
+        with open(
+            f"{join_path(self.app_interface_data_path, path)}", "r", encoding="utf-8"
+        ) as f:
+            return f.read()
+
+
+def unpack_static_variables(
+    collection_variables: TemplateCollectionVariablesV1,
+) -> dict:
+    variables = {}
+    if collection_variables.static:
+        variables = collection_variables.static
+    return variables
+
+
+def unpack_dynamic_variables(
+    collection_variables: TemplateCollectionVariablesV1,
+) -> dict:
+    variables: dict[str, Any] = {}
+    if collection_variables.dynamic:
+        gql_no_validation = init_from_config(validate_schemas=False)
+        for dv in collection_variables.dynamic or []:
+            variables[dv.name] = []
+            result = gql_no_validation.query(dv.query)
+            for key in result or {}:
+                variables[dv.name].append(result[key])
+    return variables
```

Comment:
There is no duplicate `dv.name` for one collection, right? Then it can be simplified with guard exit and comprehension

```python
    if not collection_variables.dynamic:
        return {}

    gql_no_validation = init_from_config(validate_schemas=False)

    return {
        dv.name: list((gql_no_validation.query(dv.query) or {}).values())
        for dv in collection_variables.dynamic or []
    }
```